tests/model_registry/rest_api/conftest.py

```diff
@@ -63,3 +92,250 @@ def updated_model_registry_resource(
+    volumes_mounts.extend([
+        {"name": CA_CONFIGMAP_NAME, "mountPath": f"{CA_MOUNT_PATH}/ca", "readOnly": True},
+        {
+            "name": "mysql-server-cert",
+            "mountPath": f"{CA_MOUNT_PATH}/server_cert",
+            "readOnly": True,
+        },
+        {
+            "name": "mysql-server-key",
+            "mountPath": f"{CA_MOUNT_PATH}/server_key",
+            "readOnly": True,
+        },
+    ])
+
+    my_sql_container["args"] = mysql_args
+    my_sql_container["volumeMounts"] = volumes_mounts
+    volumes = list(spec["volumes"])
+    volumes.extend([
+        {"name": CA_CONFIGMAP_NAME, "configMap": {"name": CA_CONFIGMAP_NAME}},
+        {"name": "mysql-server-cert", "secret": {"secretName": "mysql-server-cert"}},  # pragma: allowlist secret
+        {"name": "mysql-server-key", "secret": {"secretName": "mysql-server-key"}},  # pragma: allowlist secret
+    ])
+
+    patch = {"spec": {"template": {"spec": {"volumes": volumes, "containers": [my_sql_container]}}}}
+    with ResourceEditor(patches={deployment: patch}):
+        deployment.wait_for_condition(condition="Available", status="True")
+        yield deployment
+
+
+@pytest.fixture(scope="class")
+def mysql_ssl_artifacts_and_secrets(
+    admin_client: DynamicClient,
+    model_registry_namespace: str,
+) -> Generator[dict[str, Any], None, None]:
+    """
+    Generates MySQL SSL artifacts and creates corresponding Kubernetes secrets for use in tests.
+    This fixture provides file paths and secret objects for the CA certificate, server certificate, and server key.
+
+    Args:
+        admin_client: The Kubernetes dynamic client used to create secrets.
+        model_registry_namespace: The namespace in which to create the secrets.
+
+    Yields:
+        dict: A dictionary containing file paths and secret objects for the CA, server certificate, and server key.
+    """
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        paths = generate_ca_and_server_cert(tmp_dir=tmp_dir)
+
+        def create_secret(name: str, file_path: str, key_name: str) -> Secret:
+            with open(file_path, "rb") as f:
+                file_content_raw_bytes = f.read()
+                file_content_b64_string = base64.b64encode(file_content_raw_bytes).decode("utf-8")
+                data = {key_name: file_content_b64_string}
+            secret = Secret(
+                client=admin_client,
+                name=name,
+                namespace=model_registry_namespace,
+                data_dict=data,
+                wait_for_resource=True,
+            )
+            secret.create()
+            return secret
+
+        ca_secret = create_secret(name="mysql-ca", file_path=paths["ca_crt"], key_name="ca.crt")
+        server_cert_secret = create_secret(name="mysql-server-cert", file_path=paths["server_crt"], key_name="tls.crt")
+        server_key_secret = create_secret(name="mysql-server-key", file_path=paths["server_key"], key_name="tls.key")
```

Comment:
This fixture is doing too many things at a time. How about having smaller fixtures. They are typically supposed to do one thing, that way, triage is easy.

Reply:
splitted